Tests/test_file_tiff.py

```diff
@@ -450,6 +450,26 @@ def test_save_tiff_with_jpegtables(self):
         # Should not raise UnicodeDecodeError or anything else
         im.save(outfile)
 
+    def test_save_greyscale_uint16_tiff(self):
+        # Open a greyscale uint16 TIFF
+        infile = "Tests/images/uint16_greyscale.tif"
+        im = Image.open(infile)
+
+        # Saving it straight back as a uint16 greyscale TIFF shouldn't throw an exception
+        outfile = self.tempfile("temp.tif")
+        im.save(outfile)
+
+        # And then all it's parameters should be the same as the original.
+        # We don't just compare the images because the 'readonly' value here will be different, so
+        # we instead compare all other attributes.
+        out_im = Image.open(outfile)
+        self.assertEqual(im.mode, out_im.mode)
+        self.assertEqual(im.size, out_im.size)
+        self.assertEqual(im.info, out_im.info)
+        self.assertEqual(im.category, out_im.category)
+        self.assertEqual(im.getpalette(), out_im.getpalette())
+        self.assertEqual(im.tobytes(), out_im.tobytes())
+
```

Comment:
Why does numpy need to be involved? This is a uint16 TIFF that's being resaved. This test failed previously because it both threw a TypeError saying it could save uint16 TIFFs and because the data it saved was wrong, it only saved the image as a uint8 image. This test checks for both of those things.